### PR TITLE
Added attack/wound buttons to battle UI

### PR DESCRIPTION
**Pull request by KarateSnoopy**

### CODE REVIEW

File: BattleUi.ttslua [hunk 1]
```diff
@@ -165,6 +180,62 @@ function BattleUi.InitUi(ui)
             player.weaponHitTexts[i]   = Ui.Text(player.panel, { id = "WeaponHit"..i,   rectAlignment = "UpperLeft", x = 288, y = y, width = 20, height = 25, color = TEXT_COLOR, fontSize = 16, text = "44" })
             player.weaponWoundTexts[i] = Ui.Text(player.panel, { id = "WeaponWound"..i, rectAlignment = "UpperLeft", x = 334, y = y, width = 55, height = 25, color = TEXT_COLOR, fontSize = 16, text = "44 (+44)" })
             player.weaponCritTexts[i]  = Ui.Text(player.panel, { id = "WeaponCrit"..i,  rectAlignment = "UpperLeft", x = 407, y = y, width = 20, height = 25, color = TEXT_COLOR, fontSize = 16, text = "44" })
+
+            player.attackWeaponImages[i] = Ui.Image(player.panel, { id = "AttackWeaponImage"..i, rectAlignment = "UpperLeft", x = 440, y = y, width = 50, height = 25, image = "BR_Sleep" })
```
Comment (KarateSnoopy) [inline BattleUi.ttslua:184]: Each of the new button images is still using "BR_Sleep" so the the icon is wrong, but easy to fix with better art.

**Comment by KarateSnoopy** [comment]
Requires a few edits to the TTS board such as new dice rollers.  You can see those here:

https://drive.google.com/file/d/1aC8JJ18JzntWb65wNNeR1b8XKmS8-rJ6/view?usp=sharing

**Comment by MaximeLizotte** [comment]
The script breaks when there is no white coloured player. also, players can only be either GM or Spectator

**Comment by jordanmchavez** [comment]
Sorry, I had email notifications turned off and wasn't expecting any pull reqs.

This is an interesting idea, but there are some problems with this approach:
* The GUIDs shouldn't live here, but in NamedObject.
* There's code in the die roller objects themselves. A big part of this mod's load time and simplicity is having all of the code be in one place (Global).
* The die rollers don't match the design aesthetic. I'm not a fan of die rollers in general, but if I've come around a bit and have considered adding them. If they're going to be added in the mod though, I'd to redesign the board's layout to accommodate them, create the appropriate die roll arenas, add them to the location/zone system, etc. This is the biggest problem.
* I'd need to the new battle UI design/icons before anything. It's already packed pretty tight as is.
* Any kind of auto attack calculation would need to take blind, etc into account automatically (without a separate button) to minimize the interface complexity. That means a positioning system (something that I've been thinking about anyway since it wouldn't be hard to set up Zones for.

One of the important qualities of this mod is that I'm not anything that doesn't fit seamlessly with the existing visual design, so I'm not throwing anything in without careful consideration. For this to work I'd need:
* A mockup of the battle UI.
* The object and textures for the die rolling arena. This would probably be a stand-alone 3d object so people could move it around and have it be close to the fight.
* I'd need to review the visual design first to make sure it fits.
* Adding these in with the code issues above addressed.

The other problem is the assets aren't available...I can share a folder with you if you're interested, but this would be a lot of work.

**Pull request closed without merging**